sbi/samplers/mcmc/pymc_utils.py

```diff
@@ -0,0 +1,173 @@
+from typing import Callable, Optional
+
+import numpy as np
+import pymc
+import pytensor.tensor as pt
+import torch
+
+from sbi.inference.potentials.base_potential import BasePotential
+from sbi.utils import tensor2numpy
+
+
+class PyMCPotential(pt.Op):
+    itypes = [pt.dvector]  # expects a vector of parameter values when called
+    otypes = [
+        pt.dscalar,
+        pt.dvector,
+    ]  # outputs a single scalar value (the potential) and gradients for every input
+    default_output = 0  # return only potential by default
+
+    def __init__(
+        self,
+        potential_fn: BasePotential,
+        device: str,
+        track_gradients: bool = True,
+    ):
+        """PyTensor operation wrapping a `BasePotential` callable for use
+        with PyMC samplers.
+
+        Args:
+            potential_fn: Potential function that returns a potential given parameters
+            device: The device to which to move the parameters before evaluation.
+            track_gradients: Whether to track gradients from potential function
+        """
+        self.potential_fn = potential_fn
+        self.device = device
+        self.track_gradients = track_gradients
+
+    def perform(self, node, inputs, outputs):
```

Comment:
so in this function you needed to detach or copy the `params` right? could you add a short comment in the code on that?